### PR TITLE
Label and type documents

### LINKED ISSUE #6: Label documents
Users should be able to give a label for documents they upload: what kind of document is it and what are relevant topics?

### LINKED ISSUE #7: Filter documents
Users should be able to filter documents to easily find relevant documents.

### PR DESCRIPTION
**Pull request by janjanjan20**
The filter works on types and labels, must have 1 of the given types and 1 of the given labels (unless none are given). Closes #7. Closes #6.

### CODE REVIEW

File: src/app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -86,6 +94,63 @@ public function destroy(Post $post): RedirectResponse
...
+                    ->whereIn('post_type.type_id', $request->type_ids)
+                    ->whereIn('label_post.label_id', $request->label_ids)
+                    ->get();
+            }
+        }
+
+        $new_arr = Arr::pluck($wanted_posts, 'post_id');
+
+        $all_posts = Post::with('user:id,name', 'labels:name', 'types:name')->latest();
+        // $test2 = $test->whereIn('id', [26, 27])->get();
```
Comment (Stefan-Vlastuin) [inline src/app/Http/Controllers/PostController.php:142]: Remove comments.

File: src/resources/js/Layouts/AuthenticatedLayout.jsx [hunk 2]
```diff
@@ -27,6 +27,12 @@ export default function Authenticated({ user, header, children }) {
                                 <NavLink href={route('posts.index')} active={route().current('posts.index')}>
                                     Posts
                                 </NavLink>
+                                <NavLink href={route('filter')} active={route().current('filter')}>
```
Comment (Stefan-Vlastuin) [inline src/resources/js/Layouts/AuthenticatedLayout.jsx:30]: I'd say only an admin can add labels and types, right?

File: src/routes/web.php [hunk 3]
```diff
@@ -31,6 +34,50 @@
...
+Route::resource('types', TypeController::class)
+    ->only(['index', 'store'])
+    ->middleware(['auth', 'verified']);
+
+
+Route::resource('labels', LabelController::class)
+->only(['store'])
+->middleware(['auth', 'verified']);
+
+// Route::resource('filter/{types}/{labels}', [PostController::class, 'filter'])
```
Comment (Stefan-Vlastuin) [inline src/routes/web.php:57]: Remove comments.

**Review by Stefan-Vlastuin — APPROVED** [review]

File: src/resources/js/Layouts/AuthenticatedLayout.jsx (hunk 2 above)
Comment (janjanjan20) [reply src/resources/js/Layouts/AuthenticatedLayout.jsx:30]: I will implement that yes, I wanted to have what I already have merged into main already though. Before my branch becomes to distant.

**Pull request merged**